### PR TITLE
ci: don't skip buildkite steps if not a PR

### PR DESCRIPTION
**Pull request by kim**
Also, GHA seems to not correctly evaluate `== true|false`.

### CODE REVIEW

**Comment by kim** [comment]
je ne comprend pas les booleans

File: .buildkite/pipeline.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ steps:
   # still saves about 20sec to pin the image to a SHA256 after an upgrade.
   - label: "Build container"
     # Nb: `if` skips the step if the expression evaluates to false
-    if: build.pull_request.repository.fork == false
+    if: build.pull_request.repository.fork != true
```
Comment (FintanH) [inline .buildkite/pipeline.yaml:7]: Wait... WHAT?!

Comment (kim) [reply .buildkite/pipeline.yaml]: Please consider donating to "ASSOCIATION OF COMPUTER SCIENTISTS AND PRACTICIONERS FOR THE ABOLISHMENT OF YAML AND BOOLEAN LOGIC" (AOCSAPFTAOYABL)

Comment (FintanH) [reply .buildkite/pipeline.yaml]: I'll donate some RAD

**Comment by kim** [comment]
I think this does actually work now, but the checks display on the front matter gets confused because the companion PR has the same SHA

File: .github/workflows/fast.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ name: fast
 on: [pull_request]
 jobs:
   fmt:
-    if: github.event.pull_request.head.repo.fork == true
+    if: ${{ github.event.pull_request.head.repo.fork }}
```
Comment (FintanH) [inline .github/workflows/fast.yaml:5]: Assuming `null` means `false` here :smile:

**Review by FintanH — APPROVED** [review]

**Comment by FintanH** [comment]
> I think this does actually work now, but the checks display on the front matter gets confused because the companion PR has the same SHA

You could try test it by opening a PR from a fork? :shrug:

**Comment by kim** [comment]
> You could try test it by opening a PR from a fork?

That's what I did: #601 

If you switch to the "Checks" tab, you'll see two actions, one from this branch and the other from the fork. This branch correctly skipped the redundant jobs, the other one ran them. Supposedly because both PRs have the same SHA, the wrong checks are shown on the PR front page.

**Comment by kim** [comment]
Oh, and the BK check shown on the fork PR links to the build for THIS PR. I guess because of the same reason.

**Pull request merged**